### PR TITLE
[TRA-126] Add E2E test for isolated subaccount transfers.

### PR DESCRIPTION
**Pull request by vincentwschau**
### Changelist
Add E2E test for transfers from / to isolated subaccounts.

Additional changes:
- cleaned up some tech debt I found where subaccounts with no perpetual positions should have `nil` and not an empty slice for test constants
- cleaned up some tech debt I found where perpetual positions should be initialized with a 0 funding index rather than the default value for the byte array for test constants

### Test Plan
Ran test.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/testutil/constants/subaccounts.go [hunk 1]
```diff
@@ -80,8 +108,9 @@ var (
 		},
 		PerpetualPositions: []*satypes.PerpetualPosition{
 			{
-				PerpetualId: 0,
-				Quantums:    dtypes.NewInt(-10_000_000_000), // -100 BTC
+				PerpetualId:  0,
+				Quantums:     dtypes.NewInt(-10_000_000_000), // -100 BTC
+				FundingIndex: dtypes.NewInt(0),
```
Comment (shrenujb) [inline protocol/testutil/constants/subaccounts.go:113]: Why were all of these changed? Doesn't seem related to this PR?

File: protocol/x/sending/e2e/isolated_subaccount_transfers_test.go [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+			expectedCollateralPoolBalances: map[string]int64{
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 9_900_000_000, // $9,900 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.Iso2Usd_IsolatedMarket.Params.Id),
+				).String(): 10_100_000_000, // $10,100 USDC
+			},
+			expectedErr: "",
+		},
```
Comment (shrenujb) [inline protocol/x/sending/e2e/isolated_subaccount_transfers_test.go:154]: Another possible test case?
- collateral transfer from isolated subaccount to isolated subaccount with the same isolated perpetual, doesn't result in underlying transfer of collateral pool

File: protocol/testutil/constants/subaccounts.go (hunk 1 above)
Comment (vincentwschau) [reply protocol/testutil/constants/subaccounts.go:113]: I have it in the description:
>cleaned up some tech debt I found where perpetual positions should be initialized with a 0 funding index rather than the default value for the byte array for test constants

Mainly did this because I had to change it for the constants I used, which made the constants file inconsistent.

File: protocol/x/sending/e2e/isolated_subaccount_transfers_test.go (hunk 2 above)
Comment (vincentwschau) [reply protocol/x/sending/e2e/isolated_subaccount_transfers_test.go:154]: Good idea, added.

**Review by shrenujb — APPROVED** [review]
